ide/lsp.client/src/org/netbeans/modules/lsp/client/debugger/LineBreakpointData.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.lsp.client.debugger;
+
+public record LineBreakpointData(String url, int lineNumber, String condition) {
```

Comment:
Is the DAP protocol requiring that source locations are URLs? I ask because I learned the hard way, that for example the Chrome DevTools Protocol (Googles Protocol for remote control of Chrome and usable with node) used URL in many places where they meant "random string, you should not mess with, it might be an URL, but also just an internal identifier".

Reply:
My understanding is that DAP provides an option to use either paths or URIs, and paths are the default, and are currently used by this debugger. But, on this place, it is really a URL, which is then converted using the `URLPathConvertor` to the paths used by the protocol.

Reply:
Further investigation available at https://github.com/apache/netbeans/pull/8098#issuecomment-2571672773